src/world/shipparts/shield.lua

```diff
@@ -0,0 +1,88 @@
+local PhysicsReferences = require("world/physicsReferences")
+local Timer = require("timer")
+
+local Shield = class()
+
+function Shield:__create(body)
+	self.partLocations = {}
+	self.center = {0, 0}
+	self.body = body
+	self.health = 10
+	self.collidedFixtures = {}
+	self.timer = Timer(5)
+end
+
+function Shield:createFixture()
+	if self.fixture then self.fixture:destroy() end
+	if next(self.partLocations) == nil then
+		self.fixture = nil
+		return
+	end
+	x, y = unpack(self.center)
+	local shape = love.physics.newCircleShape(x, y, 6)
+	self.fixture = love.physics.newFixture(self.body, shape)
+	self.fixture:setUserData(self)
+	PhysicsReferences.setFixtureType(self.fixture, "shield")
+end
+
+function Shield:addPart(part)
+	self.partLocations[part] = {part.location[1], part.location[2]}
+	self:createFixture()
+end
+
+function Shield:removePart(part)
+	self.partLocations[part] = nil
+	self:createFixture()
+end
+
+function Shield:collision(_, fixture)
+	self.collidedFixtures[fixture] = self:test(fixture)
+end
+
+function Shield:endCollision(_, fixture)
+	self.collidedFixtures[fixture] = nil
+end
+
+function Shield:test(fixture)
+	local x, y = self.body:getWorldPoints(unpack(self.center))
+	local radius = 5 > self.health and self.health or 5
+	local fixtureX, fixtureY = fixture:getBody():getPosition()
+	local dx = fixtureX - x
+	local dy = fixtureY - y
+	return (dx * dx) + (dy * dy) < radius * radius
```

Comment:
Just a note, lume has a distance function built in, so anywhere we write this in our code would be shortened to `lume.distance(fixtureX, fixtureY, x, y)`. It even takes an optional 5th element that causes it to return the distance squared when true.

Reply:
I have no interest in using someone elses distance function.